addons/web/static/src/views/fields/properties/property_tags.js

```diff
@@ -0,0 +1,237 @@
+/** @odoo-module **/
+
+import { _lt } from "@web/core/l10n/translation";
+import { useService } from "@web/core/utils/hooks";
+import { TagsList } from "@web/views/fields/many2many_tags/tags_list";
+import { ColorList } from "@web/core/colorlist/colorlist";
+import { usePopover } from "@web/core/popover/popover_hook";
+import { AutoComplete } from "@web/core/autocomplete/autocomplete";
+import { sprintf } from "@web/core/utils/strings";
+
+const { Component } = owl;
+
+class PropertyTagsColorListPopover extends Component {}
+PropertyTagsColorListPopover.template = 'web.PropertyTagsColorListPopover';
+PropertyTagsColorListPopover.components = {
+    ColorList,
+};
+
+export class PropertyTags extends Component {
+    setup() {
+        this.notification = useService('notification');
+        this.popover = usePopover();
+    }
+
+    /**
+     * Return the list containing tags values and actions for the TagsList component.
+     */
+    get tagListItems() {
+        if (!this.props.value || !this.props.value.length) {
+            return [];
+        }
+
+        // ['a', 'b'] =>  [['a', 'A', 5], ['b', 'B', 6]]
+        const value = this.props.tags.filter(tag => this.props.value.indexOf(tag[0]) >= 0);
+
+        const canDeleteTag = !this.props.readonly && this.props.canChangeTags;
+
+        return value.map(tag => {
+            return {
+                'id': tag[0],
+                'text': tag[1],
+                'colorIndex': tag[2] || 0,
+                'onClick': (event) => this.onTagClick(event, tag[0], tag[2]),
+                'onDelete': canDeleteTag && (() => this.onTagDelete(tag[0])),
+            };
+        });
+    }
+
+    /**
+     * Return the current selected tags.
+     * Make a deep copy to not make change on the original object
+     * and to be able to discard change.
+     */
+    get currentValue() {
+        return JSON.parse(JSON.stringify(this.props.value || []));
+    }
+
+    /**
+     * Return the current tags that can be selected.
+     * Make a deep copy to not make change on the original object
+     * and to be able to discard change.
+     */
+    get currentTags() {
+        return JSON.parse(JSON.stringify(this.props.tags || []));
+    }
+
+    /**
+     * Options available in the autocomplete component.
+     */
+    get autocompleteActions() {
+        return [{
+            options: (request) => {
+                const tagsFiltered = this.props.tags.filter(tag =>
```

Comment:
Can't we use "tagListItems" so we have a nice dict instead of index based access?

(And maybe adapt tagsListItems to instead make it a method that allows converting the tags into a dict, so we can convert either this.props.value or "currentTags", not sure if that makes sense).